plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -651,23 +717,58 @@ private static void verifySupportedDate(long value)
         }
     }
 
-    private static LongWriteFunction timestampSecondsWriteFunction()
+    private static LongWriteFunction timestampSecondsWriteFunction(ClickHouseVersion clickHouseVersion)
     {
         return (statement, index, value) -> {
             long epochSecond = floorDiv(value, MICROSECONDS_PER_SECOND);
             int nanoFraction = floorMod(value, MICROSECONDS_PER_SECOND) * NANOSECONDS_PER_MICROSECOND;
             verify(nanoFraction == 0, "Nanos of second must be zero: '%s'", value);
-            verifySupportedTimestamp(epochSecond);
-            statement.setObject(index, LocalDateTime.ofEpochSecond(epochSecond, 0, UTC));
+            LocalDateTime result = LocalDateTime.ofEpochSecond(epochSecond, 0, UTC);
+            DATETIMES_RANGE_CHECKER.verifyDateTimeSupportedTimestamp(clickHouseVersion, result, -1);
```

Comment:
`-1` is not much readable, so let's introduce a constant called `DATETIME_NO_PRECISION` to make it more readable.